rs3_trading/historicaldata/request_errors.py

```diff
@@ -0,0 +1,40 @@
+from http import HTTPStatus
+
+from requests import Response
+
+# Error Classes
+
+
+class TooManyRequestsError(Exception):
+
+    def __init__(self, status_code, message='You have reached the rate limit: Too Many Requests'):
+        self.status_code = status_code
+        self.message - message
+        super().__init__(self.message)
+
+
+class NotFoundError(Exception):
+
+    def __init__(self, status_code, message='Website Content has been removed, moved, or this is an incorrect URL: Not Found'):
+        self.status_code = status_code
+        self.message - message
+        super().__init__(self.message)
+
+
+class RequestTimeoutError(Exception):
+
+    def __init__(self, status_code, message='Idle connection and the server would like to shut it down: Request Timeout'):
+        self.status_code = status_code
+        self.message - message
+        super().__init__(self.message)
+
+
+# Error Functions
+
+def check_request_status(response: Response):
+    if response.status_code == HTTPStatus.TOO_MANY_REQUESTS:
```

Comment:
Try out a switch statement here instead of the if

Reply:
match?

Reply:
Match might be the python equivalent of a switch, I think it would be more elegant